clients/spark/core/src/main/scala/io/treeverse/utils/AsInterfaceClassLoader.scala

```diff
@@ -0,0 +1,77 @@
+package io.treeverse.utils
+
+import org.objectweb.asm.{ClassReader, ClassWriter}
+import org.objectweb.asm.util.{CheckClassAdapter, TraceClassVisitor}
+import java.io.{ByteArrayOutputStream, PrintWriter}
+import com.google.common.io.ByteStreams
+
+class AsInterfaceClassLoader(
+    parent: ClassLoader,
+    ifaces: Map[String, Class[_]],
+    val pw: PrintWriter = null
+) extends ClassLoader(parent) {
+
+  val skip = ifaces.values.map(_.getName).toSet
```

Comment:
Added a comment.  These are the interfaces into which we shall translate.  It won't work to translate them when/if they appear.  And appear they will, because this ClassLoader will try to load a class that uses them (it translates to use them), and then try to load them.